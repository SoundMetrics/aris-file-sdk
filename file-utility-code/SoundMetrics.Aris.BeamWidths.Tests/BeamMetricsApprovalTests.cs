using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoundMetrics.Aris.Headers;
using System.Text;

namespace SoundMetrics.Aris.BeamWidths.Tests
{
    [TestClass]
    [UseApprovalSubdirectory("approval-files"), UseReporter(typeof(DiffReporter))]
    public sealed class BeamMetricsApprovalTests
    {
        [TestMethod]
        public void BeamWidths_ARIS3000_128()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 128, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS3000_64()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 64, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.Telephoto));

        private static void RunTest(BeamConfig cfg)
        {
            var stringOutput = BuildStringOutput(cfg);
            Approvals.Verify(stringOutput);
        }

        private static string BuildStringOutput(BeamConfig cfg)
        {
            var beamInfos = Metrics.GetBeamInformation(cfg);
            var stringOutput = BuildOutput(beamInfos);
            return stringOutput;

            static string BuildOutput(BeamInfo[] beamInfos)
            {
                var buffer = new StringBuilder();
                buffer.AppendLine($"# of BeamInfo members: [{beamInfos.Length}]");

                foreach (var beamInfo in beamInfos)
                {
                    buffer.AppendLine(beamInfo.ToString());
                }

                return buffer.ToString();
            }
        }
    }
}
