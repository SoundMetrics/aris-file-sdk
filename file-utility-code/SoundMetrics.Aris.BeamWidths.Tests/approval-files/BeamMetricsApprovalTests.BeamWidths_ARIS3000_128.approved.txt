# of BeamInfo members: [128]
BeamInfo { BeamNumber = 0, Center = -15.0068, Left = -15.1218, Right = -14.8918 }
BeamInfo { BeamNumber = 1, Center = -14.7768, Left = -14.8918, Right = -14.6615 }
BeamInfo { BeamNumber = 2, Center = -14.5462, Left = -14.6615, Right = -14.4306 }
BeamInfo { BeamNumber = 3, Center = -14.315, Left = -14.4306, Right = -14.1992 }
BeamInfo { BeamNumber = 4, Center = -14.0833, Left = -14.1992, Right = -13.9672 }
BeamInfo { BeamNumber = 5, Center = -13.8511, Left = -13.9672, Right = -13.7349 }
BeamInfo { BeamNumber = 6, Center = -13.6186, Left = -13.7349, Right = -13.5022 }
BeamInfo { BeamNumber = 7, Center = -13.3858, Left = -13.5022, Right = -13.2693 }
BeamInfo { BeamNumber = 8, Center = -13.1528, Left = -13.2693, Right = -13.0362 }
BeamInfo { BeamNumber = 9, Center = -12.9196, Left = -13.0362, Right = -12.8029 }
BeamInfo { BeamNumber = 10, Center = -12.6861, Left = -12.8029, Right = -12.5692 }
BeamInfo { BeamNumber = 11, Center = -12.4523, Left = -12.5692, Right = -12.3353 }
BeamInfo { BeamNumber = 12, Center = -12.2182, Left = -12.3353, Right = -12.101 }
BeamInfo { BeamNumber = 13, Center = -11.9838, Left = -12.101, Right = -11.8665 }
BeamInfo { BeamNumber = 14, Center = -11.7491, Left = -11.8665, Right = -11.6316 }
BeamInfo { BeamNumber = 15, Center = -11.5141, Left = -11.6316, Right = -11.3965 }
BeamInfo { BeamNumber = 16, Center = -11.2789, Left = -11.3965, Right = -11.1612 }
BeamInfo { BeamNumber = 17, Center = -11.0435, Left = -11.1612, Right = -10.9257 }
BeamInfo { BeamNumber = 18, Center = -10.8079, Left = -10.9257, Right = -10.69 }
BeamInfo { BeamNumber = 19, Center = -10.5721, Left = -10.69, Right = -10.4541 }
BeamInfo { BeamNumber = 20, Center = -10.3361, Left = -10.4541, Right = -10.218 }
BeamInfo { BeamNumber = 21, Center = -10.0999, Left = -10.218, Right = -9.9817 }
BeamInfo { BeamNumber = 22, Center = -9.8635, Left = -9.9817, Right = -9.7452 }
BeamInfo { BeamNumber = 23, Center = -9.6269, Left = -9.7452, Right = -9.5086 }
BeamInfo { BeamNumber = 24, Center = -9.3902, Left = -9.5086, Right = -9.2718 }
BeamInfo { BeamNumber = 25, Center = -9.1534, Left = -9.2718, Right = -9.035 }
BeamInfo { BeamNumber = 26, Center = -8.9165, Left = -9.035, Right = -8.798 }
BeamInfo { BeamNumber = 27, Center = -8.6795, Left = -8.798, Right = -8.561 }
BeamInfo { BeamNumber = 28, Center = -8.4424, Left = -8.561, Right = -8.3239 }
BeamInfo { BeamNumber = 29, Center = -8.2053, Left = -8.3239, Right = -8.0868 }
BeamInfo { BeamNumber = 30, Center = -7.9682, Left = -8.0868, Right = -7.8496 }
BeamInfo { BeamNumber = 31, Center = -7.731, Left = -7.8496, Right = -7.6124 }
BeamInfo { BeamNumber = 32, Center = -7.4938, Left = -7.6124, Right = -7.3752 }
BeamInfo { BeamNumber = 33, Center = -7.2566, Left = -7.3752, Right = -7.138 }
BeamInfo { BeamNumber = 34, Center = -7.0193, Left = -7.138, Right = -6.9007 }
BeamInfo { BeamNumber = 35, Center = -6.782, Left = -6.9007, Right = -6.6633 }
BeamInfo { BeamNumber = 36, Center = -6.5446, Left = -6.6633, Right = -6.4259 }
BeamInfo { BeamNumber = 37, Center = -6.3072, Left = -6.4259, Right = -6.1885 }
BeamInfo { BeamNumber = 38, Center = -6.0698, Left = -6.1885, Right = -5.9511 }
BeamInfo { BeamNumber = 39, Center = -5.8324, Left = -5.9511, Right = -5.7137 }
BeamInfo { BeamNumber = 40, Center = -5.5949, Left = -5.7137, Right = -5.4762 }
BeamInfo { BeamNumber = 41, Center = -5.3574, Left = -5.4762, Right = -5.2387 }
BeamInfo { BeamNumber = 42, Center = -5.1199, Left = -5.2387, Right = -5.0011 }
BeamInfo { BeamNumber = 43, Center = -4.8823, Left = -5.0011, Right = -4.7635 }
BeamInfo { BeamNumber = 44, Center = -4.6447, Left = -4.7635, Right = -4.5259 }
BeamInfo { BeamNumber = 45, Center = -4.4071, Left = -4.5259, Right = -4.2883 }
BeamInfo { BeamNumber = 46, Center = -4.1695, Left = -4.2883, Right = -4.0507 }
BeamInfo { BeamNumber = 47, Center = -3.9318, Left = -4.0507, Right = -3.813 }
BeamInfo { BeamNumber = 48, Center = -3.6941, Left = -3.813, Right = -3.5753 }
BeamInfo { BeamNumber = 49, Center = -3.4564, Left = -3.5753, Right = -3.3376 }
BeamInfo { BeamNumber = 50, Center = -3.2187, Left = -3.3376, Right = -3.0998 }
BeamInfo { BeamNumber = 51, Center = -2.9809, Left = -3.0998, Right = -2.862 }
BeamInfo { BeamNumber = 52, Center = -2.743, Left = -2.862, Right = -2.624 }
BeamInfo { BeamNumber = 53, Center = -2.505, Left = -2.624, Right = -2.386 }
BeamInfo { BeamNumber = 54, Center = -2.2669, Left = -2.386, Right = -2.1478 }
BeamInfo { BeamNumber = 55, Center = -2.0287, Left = -2.1478, Right = -1.9096 }
BeamInfo { BeamNumber = 56, Center = -1.7904, Left = -1.9096, Right = -1.6712 }
BeamInfo { BeamNumber = 57, Center = -1.552, Left = -1.6712, Right = -1.4328 }
BeamInfo { BeamNumber = 58, Center = -1.3135, Left = -1.4328, Right = -1.1942 }
BeamInfo { BeamNumber = 59, Center = -1.0749, Left = -1.1942, Right = -0.9556 }
BeamInfo { BeamNumber = 60, Center = -0.8362, Left = -0.9556, Right = -0.7168 }
BeamInfo { BeamNumber = 61, Center = -0.5974, Left = -0.7168, Right = -0.478 }
BeamInfo { BeamNumber = 62, Center = -0.3585, Left = -0.478, Right = -0.2391 }
BeamInfo { BeamNumber = 63, Center = -0.1196, Left = -0.2391, Right = -0.0001 }
BeamInfo { BeamNumber = 64, Center = 0.1194, Left = -0.0001, Right = 0.2389 }
BeamInfo { BeamNumber = 65, Center = 0.3584, Left = 0.2389, Right = 0.4779 }
BeamInfo { BeamNumber = 66, Center = 0.5973, Left = 0.4779, Right = 0.7168 }
BeamInfo { BeamNumber = 67, Center = 0.8362, Left = 0.7168, Right = 0.9556 }
BeamInfo { BeamNumber = 68, Center = 1.075, Left = 0.9556, Right = 1.1944 }
BeamInfo { BeamNumber = 69, Center = 1.3137, Left = 1.1944, Right = 1.433 }
BeamInfo { BeamNumber = 70, Center = 1.5523, Left = 1.433, Right = 1.6716 }
BeamInfo { BeamNumber = 71, Center = 1.7908, Left = 1.6716, Right = 1.91 }
BeamInfo { BeamNumber = 72, Center = 2.0292, Left = 1.91, Right = 2.1484 }
BeamInfo { BeamNumber = 73, Center = 2.2675, Left = 2.1484, Right = 2.3866 }
BeamInfo { BeamNumber = 74, Center = 2.5057, Left = 2.3866, Right = 2.6248 }
BeamInfo { BeamNumber = 75, Center = 2.7438, Left = 2.6248, Right = 2.8628 }
BeamInfo { BeamNumber = 76, Center = 2.9818, Left = 2.8628, Right = 3.1008 }
BeamInfo { BeamNumber = 77, Center = 3.2197, Left = 3.1008, Right = 3.3386 }
BeamInfo { BeamNumber = 78, Center = 3.4575, Left = 3.3386, Right = 3.5764 }
BeamInfo { BeamNumber = 79, Center = 3.6952, Left = 3.5764, Right = 3.8141 }
BeamInfo { BeamNumber = 80, Center = 3.9329, Left = 3.8141, Right = 4.0518 }
BeamInfo { BeamNumber = 81, Center = 4.1706, Left = 4.0518, Right = 4.2895 }
BeamInfo { BeamNumber = 82, Center = 4.4083, Left = 4.2895, Right = 4.5271 }
BeamInfo { BeamNumber = 83, Center = 4.6459, Left = 4.5271, Right = 4.7647 }
BeamInfo { BeamNumber = 84, Center = 4.8835, Left = 4.7647, Right = 5.0023 }
BeamInfo { BeamNumber = 85, Center = 5.1211, Left = 5.0023, Right = 5.2399 }
BeamInfo { BeamNumber = 86, Center = 5.3587, Left = 5.2399, Right = 5.4775 }
BeamInfo { BeamNumber = 87, Center = 5.5962, Left = 5.4775, Right = 5.715 }
BeamInfo { BeamNumber = 88, Center = 5.8337, Left = 5.715, Right = 5.9525 }
BeamInfo { BeamNumber = 89, Center = 6.0712, Left = 5.9525, Right = 6.1899 }
BeamInfo { BeamNumber = 90, Center = 6.3086, Left = 6.1899, Right = 6.4273 }
BeamInfo { BeamNumber = 91, Center = 6.546, Left = 6.4273, Right = 6.6647 }
BeamInfo { BeamNumber = 92, Center = 6.7834, Left = 6.6647, Right = 6.9021 }
BeamInfo { BeamNumber = 93, Center = 7.0208, Left = 6.9021, Right = 7.1395 }
BeamInfo { BeamNumber = 94, Center = 7.2581, Left = 7.1395, Right = 7.3768 }
BeamInfo { BeamNumber = 95, Center = 7.4954, Left = 7.3768, Right = 7.614 }
BeamInfo { BeamNumber = 96, Center = 7.7326, Left = 7.614, Right = 7.8512 }
BeamInfo { BeamNumber = 97, Center = 7.9698, Left = 7.8512, Right = 8.0884 }
BeamInfo { BeamNumber = 98, Center = 8.207, Left = 8.0884, Right = 8.3256 }
BeamInfo { BeamNumber = 99, Center = 8.4441, Left = 8.3256, Right = 8.5627 }
BeamInfo { BeamNumber = 100, Center = 8.6812, Left = 8.5627, Right = 8.7998 }
BeamInfo { BeamNumber = 101, Center = 8.9183, Left = 8.7998, Right = 9.0368 }
BeamInfo { BeamNumber = 102, Center = 9.1553, Left = 9.0368, Right = 9.2738 }
BeamInfo { BeamNumber = 103, Center = 9.3922, Left = 9.2738, Right = 9.5106 }
BeamInfo { BeamNumber = 104, Center = 9.629, Left = 9.5106, Right = 9.7474 }
BeamInfo { BeamNumber = 105, Center = 9.8657, Left = 9.7474, Right = 9.984 }
BeamInfo { BeamNumber = 106, Center = 10.1023, Left = 9.984, Right = 10.2205 }
BeamInfo { BeamNumber = 107, Center = 10.3387, Left = 10.2205, Right = 10.4568 }
BeamInfo { BeamNumber = 108, Center = 10.5749, Left = 10.4568, Right = 10.6929 }
BeamInfo { BeamNumber = 109, Center = 10.8109, Left = 10.6929, Right = 10.9288 }
BeamInfo { BeamNumber = 110, Center = 11.0467, Left = 10.9288, Right = 11.1645 }
BeamInfo { BeamNumber = 111, Center = 11.2823, Left = 11.1645, Right = 11.4 }
BeamInfo { BeamNumber = 112, Center = 11.5177, Left = 11.4, Right = 11.6353 }
BeamInfo { BeamNumber = 113, Center = 11.7529, Left = 11.6353, Right = 11.8704 }
BeamInfo { BeamNumber = 114, Center = 11.9879, Left = 11.8704, Right = 12.1053 }
BeamInfo { BeamNumber = 115, Center = 12.2226, Left = 12.1053, Right = 12.3398 }
BeamInfo { BeamNumber = 116, Center = 12.457, Left = 12.3398, Right = 12.5741 }
BeamInfo { BeamNumber = 117, Center = 12.6911, Left = 12.5741, Right = 12.808 }
BeamInfo { BeamNumber = 118, Center = 12.9249, Left = 12.808, Right = 13.0417 }
BeamInfo { BeamNumber = 119, Center = 13.1584, Left = 13.0417, Right = 13.275 }
BeamInfo { BeamNumber = 120, Center = 13.3916, Left = 13.275, Right = 13.5081 }
BeamInfo { BeamNumber = 121, Center = 13.6246, Left = 13.5081, Right = 13.741 }
BeamInfo { BeamNumber = 122, Center = 13.8574, Left = 13.741, Right = 13.9737 }
BeamInfo { BeamNumber = 123, Center = 14.0899, Left = 13.9737, Right = 14.206 }
BeamInfo { BeamNumber = 124, Center = 14.3221, Left = 14.206, Right = 14.438 }
BeamInfo { BeamNumber = 125, Center = 14.5538, Left = 14.438, Right = 14.6694 }
BeamInfo { BeamNumber = 126, Center = 14.785, Left = 14.6694, Right = 14.9003 }
BeamInfo { BeamNumber = 127, Center = 15.0156, Left = 14.9003, Right = 15.1306 }
